packages/core/src/lib/TwaManifest.ts

```diff
@@ -315,6 +315,101 @@ export class TwaManifest {
     const json = JSON.parse((await fs.promises.readFile(fileName)).toString());
     return new TwaManifest(json);
   }
+
+  /**
+   * Given a field name, returns the new value of the field
+   *
+   * @param {string} fieldName the name of the given field.
+   * @param {string[]} fieldsToIgnore the fields which needs to be ignored.
+   * @param {T} oldValue the old value of the field.
+   * @param {T} newValue the new value of the field.
+   * @returns {T}
+   */
+  static getNewFieldValue<T>(fieldName: string, fieldsToIgnore: string[],
+      oldValue: T, newValue: T): T {
+    if (fieldName in fieldsToIgnore) {
+      return oldValue;
+    }
+    return newValue || oldValue;
+  }
+
+  /**
+   * Merges the Twa Manifest with the web manifest. Ignores the specified fields.
+   *
+   * @param {string[]} fieldsToIgnore the fields which needs to be ignored.
+   * @param {URL} webManifestUrl the URL where the webmanifest is available.
+   * @param {WebManifest} webManifest the Web Manifest, used as a base for the update of
+   *    the TWA Manifest.
+   * @param {TwaManifest} oldTwaManifest current Twa Manifest.
+   */
+  static merge(fieldsToIgnore: string[], webManifestUrl: URL,
+      webManifest: WebManifestJson, oldTwaManifest: TwaManifest): void {
+    function resolveIconUrl(icon: WebManifestIcon | null): string | undefined {
+      return icon ? new URL(icon.src, webManifestUrl).toString() : undefined;
+    }
+
+    let shortcuts: ShortcutInfo[] = [];
+    if (!('shortcuts' in fieldsToIgnore)) {
+      for (let i = 0; i < (webManifest.shortcuts || []).length; i++) {
+        const s = webManifest.shortcuts![i];
+        try {
+          const shortcutInfo = ShortcutInfo.fromShortcutJson(webManifestUrl, s);
+          if (shortcutInfo != null) {
+            shortcuts.push(shortcutInfo);
+          }
+        } catch (err) {
+          TwaManifest.log.warn(`Skipping shortcut[${i}] for ${err.message}.`);
+        }
+        if (shortcuts.length === 4) {
+          break;
+        }
+      }
```

Comment:
I feel like this bit of logic probably exists somewhere else already. Could we make a `function getShortcutInfos(manifest: WebManifestJson): ShortcutInfo[]` and use that in both places?